ironfish/src/wallet/account.ts

```diff
@@ -585,10 +587,20 @@ export class Account {
           continue
         }
 
-        if (!note.spent) {
-          unconfirmedCount++
-          confirmed -= note.note.value()
+        if (unconfirmedTransactionHashes.has(note.transactionHash)) {
```

Comment:
This makes the calculation of unconfirmed + confirmed balances better than it was, so I think we should merge it, but I think there's a further change needed.

It looks like this relies on the user having a note in the unconfirmed range, which in the case of this test, is happening because a change note is generated and returned to the user in the transaction spending another note. However, if the transaction spends an entire note without returning any change, then there won't be any notes in the confirmation range to trigger loading the transaction's balance deltas (only a note spend).

I don't think our fixtures allow us to change transaction amounts now, but I simulated this by editing fixtures to create a transaction that spends 2000000000n-value.

Reply:
Ah yeah we can't easily catch that case without iterating over all transactions right now 😬

I think adding an index 'sequenceToTransactionHash' will resolve this by providing a way for to us iterate over transactions in the sequence range.